packages/workmanager/tizen/src/job_scheduler_wrapper.cc

```diff
@@ -0,0 +1,227 @@
+    if (ret != JOB_ERROR_NONE) {
+      LOG_ERROR("Failed to set job info battery_not_low: %s",
+                get_error_message(ret));
+      return ret;
+    }
+  }
+
+  if (constraints.charging) {
+    ret = job_info_set_requires_charging(job_info, true);
+    if (ret != JOB_ERROR_NONE) {
+      LOG_ERROR("Failed to set job info charging: %s", get_error_message(ret));
+      return ret;
+    }
+  }
+
+  switch (constraints.network_type) {
+    case NetworkType::kConnected:
+    case NetworkType::kUnmetered:
+      ret = job_info_set_requires_wifi_connection(job_info, true);
+      if (ret != JOB_ERROR_NONE) {
+        LOG_ERROR("Failed to set job info wifi_connection: %s",
+                  get_error_message(ret));
+        return ret;
+      }
+      break;
+  }
+  return ret;
+}
+
+void JobScheduler::RegisterJob(const JobInfo& job_info,
+                               job_service_callback_s* callback) {
+  job_info_h job_handle;
+  int ret = job_info_create(&job_handle);
+  if (ret != JOB_ERROR_NONE) {
+    LOG_ERROR("Failed to create job info: %s", get_error_message(ret));
+    return;
+  }
+
+  ret = SetJobConstraints(job_handle, job_info.constraints);
+  if (ret != JOB_ERROR_NONE) {
+    job_info_destroy(job_handle);
+    return;
+  }
+
+  if (job_info.is_periodic) {
+    ret = job_info_set_periodic(job_handle, job_info.frequency_seconds / 60);
+    if (ret != JOB_ERROR_NONE) {
+      LOG_ERROR("Failed to set job info periodic: %s", get_error_message(ret));
+      job_info_destroy(job_handle);
+      return;
+    }
+    ret = job_info_set_persistent(job_handle, true);
+    if (ret != JOB_ERROR_NONE) {
+      LOG_ERROR("Failed to set job info persistent: %s",
+                get_error_message(ret));
+      job_info_destroy(job_handle);
+      return;
+    }
+  } else {
+    ret = job_info_set_once(job_handle, true);
+    if (ret != JOB_ERROR_NONE) {
+      LOG_ERROR("Failed to set job info once: %s", get_error_message(ret));
+      job_info_destroy(job_handle);
+      return;
+    }
+    ret = job_info_set_persistent(job_handle, false);
+    if (ret != JOB_ERROR_NONE) {
+      LOG_ERROR("Failed to set job non-persistent: %s", get_error_message(ret));
+      job_info_destroy(job_handle);
+      return;
+    }
+  }
+
+  ret = job_scheduler_schedule(job_handle, job_info.unique_name.c_str());
+  if (ret != JOB_ERROR_NONE) {
+    if (ret == JOB_ERROR_ALREADY_EXIST) {
+      switch (job_info.existing_work_policy) {
+        case ExistingWorkPolicy::kReplace:
+        case ExistingWorkPolicy::kUpdate:
+          CancelByUniqueName(job_info.unique_name);
+          ret =
+              job_scheduler_schedule(job_handle, job_info.unique_name.c_str());
+          if (ret != JOB_ERROR_NONE) {
+            LOG_ERROR("Failed to schedule job: %s", get_error_message(ret));
+          } else {
+            SaveJobInfo(job_info);
+            if (!callback) {
+              break;
+            }
```

Comment:
Is `SaveJobInfo` running even if it is `callback == nullptr`?

Reply:
fixed by return when callback is nullptr at the front of RegisterJob.